mage/tutorials/create-a-new-module-cpp.md

```diff
@@ -0,0 +1,217 @@
+
+  mg_utility::InsertIntValueResult(record, kFieldStep, step, memory);
+  mg_utility::InsertNodeValueResult(graph, record, kFieldNode, node_id, memory);
+```
+:::tip
+Analogous methods for other supported data types are outlined in the C API reference.
+:::
+
+The `mgp_init_module` function has as its main duty the registration of procedure(s)
+which can then be invoked in openCypher. With the C API, we add our procedure and its
+inputs and outputs.
+
+```cpp
+extern "C" int mgp_init_module(struct mgp_module *module,
+                               struct mgp_memory *memory) {
+  {
+    try {
+      auto *proc = mgp::module_add_read_procedure(module, "random_walk",
+                                                  RandomWalk);
+
+      // optional parameters require a default value
+      auto default_steps = mgp::value_make_int(10, memory);
+
+      mgp::proc_add_arg(proc, kStart, mgp::type_node());
+      mgp::proc_add_opt_arg(proc, kSteps, mgp::type_int(), default_steps);
+
+      mgp::proc_add_result(proc, kFieldRandomWalk, mgp::type_path());
+
+      mgp_value_destroy(default_steps);
+    } catch (const std::exception &e) {
+      return 1;
+    }
+  }
+  return 0;
+}
+```
+Although this example registers a single `procedure`, you can have multiple different
+procedures in a single module, each of which can be invoked using the 
+`CALL <module>.<procedure> ...` syntax (`<module>` being the name of the shared library).
+As we compile our example to `random_walk.so`, the module name is thus `random_walk`. 
+
+:::tip
+Note that, as the procedure name is defined upon registration, it can differ from its
+corresponding callback.
+:::
+
+:::info
+For more information on the signature specification API, consult the documentation of
+the functions prefixed with `proc_` in the `mgp.hpp` file.
+:::
+
+:::note
+As the `memory` argument is only alive throughout the execution of `mgp_init_module`,
+do not allocate any global resources with it. If you still do need to set up a global
+state, you may do so in the `mgp_init_module` using the standard global allocators.
+:::
+
+Finally, you may want to reset any global state or release global resources, which is 
+done in the following function:
+
+```cpp
+extern "C" int mgp_shutdown_module() {
+   return 0;
+}
+```
+
+As mentioned before, no exceptions should leave your module. As done in this example,
+exception handlers are there in `mgp_init_module` and the callback function.
+Depending on your module’s needs, you might want one in `mgp_shutdown_module` as well.
+
+## Running
+
+To compile, build and run the module, refer to the corresponding section of the Python
+tutorial (minus steps 2a-2d).
+
+## Querying
+
+As querying is independent of the query module’s implementation language, all relevant
+information is again available in the Python tutorial.
+
+## Testing
+
+The same remark holds for testing – relevant information is provided inside the Python
+tutorial.
```

Comment:
This should not refer to Python tutorial, therefore needs to be extracted: @antoniofilipovic

Reply:
Done!